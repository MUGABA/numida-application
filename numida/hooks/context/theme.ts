export type Theme = {
  background: string;
  primary: string;
  secondary: string;
  text: string;
  buttonText: string;
  border: string;
};

export const lightTheme: Theme = {
  background: "#ffffff",
  primary: "#30C2E3",
  secondary: "#D4FAF5",
  text: "",
  buttonText: "",
  border: "",
};

export const darkTheme: Theme = {
  background: "#121212",
  primary: "#bb86fc",
  secondary: "#03dac6",
  text: "#ffffff",
  buttonText: "#000000",
  border: "#333333",
};
