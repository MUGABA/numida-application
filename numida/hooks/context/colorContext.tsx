// ThemeContext.tsx
import React, { createContext, ReactNode, useContext, useState } from "react";
import { darkTheme, lightTheme, Theme } from "./theme";

interface ThemeContextProps {
  theme: Theme;
  toggleTheme: () => void;
}

const ThemeContext = createContext<ThemeContextProps | undefined>(undefined);

export const ColorThemeProvider: React.FC<{ children: ReactNode }> = ({
  children,
}) => {
  const [theme, setTheme] = useState<Theme>(lightTheme);

  const toggleTheme = () => {
    setTheme((prevTheme) =>
      prevTheme === lightTheme ? darkTheme : lightTheme
    );
  };

  return (
    <ThemeContext.Provider value={{ theme, toggleTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

export const useColorTheme = (): ThemeContextProps => {
  const context = useContext(ThemeContext);
  if (!context) {
    throw new Error("useTheme must be used within a ThemeProvider");
  }
  return context;
};
