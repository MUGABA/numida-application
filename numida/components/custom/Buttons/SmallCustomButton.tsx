import { useColorTheme } from "@/hooks/context/colorContext";
import React from "react";
import { Image, Pressable, StyleSheet, Text, View } from "react-native";

type Props = {
  buttonTitle: string;
  icon: HTMLImageElement;
};

const SmallCustomButton = ({ buttonTitle, icon }: Props) => {
  const { theme } = useColorTheme();
  return (
    <View>
      <Pressable style={{ ...styles.btn, borderColor: theme.primary }}>
        <View>
          <Text style={{ ...styles.btnTitle, color: theme.primary }}>
            {buttonTitle}
          </Text>
        </View>
        <View style={styles.icon}>
          <Image source={icon} />
        </View>
      </Pressable>
    </View>
  );
};

export default SmallCustomButton;

const styles = StyleSheet.create({
  btn: {
    width: 120,
    height: 21,
    flex: 2,
    flexDirection: "row",
    justifyContent: "space-between",
    borderWidth: 1,
    borderRadius: 10,
    marginVertical: 3,
  },
  btnTitle: {
    paddingVertical: 5,
    paddingHorizontal: 5,
  },

  icon: {
    paddingVertical: 10,
    paddingHorizontal: 3,
    fontSize: 20,
  },
});
