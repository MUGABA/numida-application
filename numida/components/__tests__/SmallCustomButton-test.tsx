import { ColorThemeProvider } from "@/hooks/context/colorContext";
import { render } from "@testing-library/react-native";
import React from "react";
import SmallCustomButton from "../custom/Buttons/SmallCustomButton";

const renderWithProvider = (ui: React.ReactElement) => {
  return render(<ColorThemeProvider>{ui}</ColorThemeProvider>);
};

describe("Small Button Component", () => {
  it("It renders button title collectly", () => {
    const { getByText } = renderWithProvider(
      <SmallCustomButton
        buttonTitle="Learn More"
        icon={require("../../assets/images/Vector.png")}
      />
    );
    expect(getByText("Learn More")).toBeTruthy();
  });

  it(`renders correctly`, () => {
    const onClickMock = jest.fn();
    const { toJSON } = render(
      <ColorThemeProvider>
        <SmallCustomButton
          buttonTitle="Submit"
          icon={require("../../assets/images/Vector.png")}
        ></SmallCustomButton>
      </ColorThemeProvider>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
