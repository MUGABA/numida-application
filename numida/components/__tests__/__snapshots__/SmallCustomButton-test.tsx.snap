// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Small Button Component renders correctly 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "borderColor": "#30C2E3",
        "borderRadius": 10,
        "borderWidth": 1,
        "flex": 2,
        "flexDirection": "row",
        "height": 21,
        "justifyContent": "space-between",
        "marginVertical": 3,
        "width": 120,
      }
    }
  >
    <View>
      <Text
        style={
          {
            "color": "#30C2E3",
            "paddingHorizontal": 5,
            "paddingVertical": 5,
          }
        }
      >
        Submit
      </Text>
    </View>
    <View
      style={
        {
          "fontSize": 20,
          "paddingHorizontal": 3,
          "paddingVertical": 10,
        }
      }
    >
      <Image
        source={1}
      />
    </View>
  </View>
</View>
`;
